### DISCUSSION TITLE
Blinking filament runout icon

### DISCUSSION THREAD

**Discussion by Crazy-Charles** [mriscoc/Ender3V2S1#946] [General]
Me again, I may have found an issue where the filament runout icon blinks regardless of the state of the "Enable Runout" setting, my fix for this in DWIN.CPP is as follows,
```cpp
#if HAS_FILAMENT_SENSOR && PROUI_EX
  void _draw_runout_icon() {
    if (runout.enabled) {
      static bool _runout_active = false;
      _draw_iconblink(_runout_active, FilamentSensorDevice::poll_runout_state(0), ICON_StepE, ICON_FilRunOut, 112, 416);
    }
    else {
     DWIN_Draw_Box(1, HMI_data.Background_Color, 112, 416, 20, 20); 
     DWINUI::Draw_Icon(ICON_StepE, 112, 416);
    }
  }
#endif
```
Seems to work ok but as I'm not a programmer it may not be the best solution

**Comment by mriscoc** [comment]
I understand your proposal and maybe your are right, if the filament runout sensor is disabled in the UI, maybe is a good idea to disable also the support for the icon blinking. The same could be in the case of the endstop diagnostic page, so could be better to change the FilamentSensorDevice::poll_runout_state(0) itself.

**Comment by mriscoc** [comment]
Latest versions disable background icon blink if the filament runout sensor is disabled.
